### ISSUE TITLE
📝 Correction | Encrypted DNS can be anonymous with DNSCrypt v2 protocol

### ISSUE THREAD

**Issue by lrq3000** [privacytools/privacytools.io#1822]
## Description

While writing #1821, I have just discovered that the DNSCrypt v2 protocol supports anonymized DNS queries. Here is the [documentation](https://github.com/DNSCrypt/dnscrypt-protocol/blob/master/ANONYMIZED-DNSCRYPT.txt).

It seems to only be implemented in one software at the moment, in [dnscrypt-proxy](https://github.com/DNSCrypt/dnscrypt-proxy).

Maybe other softwares such as [Simple DNSCrypt](https://simplednscrypt.org/), which is based on dnscrypt-proxy, may implement (or will in the future) anonymized DNS queries, but at the moment it seems from [this listing](https://dnscrypt.info/implementations/) that it's not the case.

## Why I am making the suggestion

In https://www.privacytools.io/providers/dns/ , it's written:

> Note: Using an encrypted DNS resolver will not make you anonymous

DNSCrypt v2 seems to fix this issue, and there is a concrete application.

## My connection with the software

- [x] I will keep the issue up-to-date if something I have said changes or I remember a connection with the software.

**Comment by beerisgood** [comment]
Well that's not fully true. It exist exist the eSNI problem.

**Comment by lrq3000** [comment]
Very interesting, I admit I have no expertise in DNS and such, I am more of a savvy end-user (I can adopt a new protocol/framework but I could certainly not tweak it nor make one).

I found this article about [eSNI](https://www.sentinelone.com/blog/privacy-2019-fixing-16-year-old-problem-tls-esni/), is this what you were referring to? From the doc, I understand that SNI had an issue with privacy, but eSNI should solve this. However, I did not find any mention about eSNI on the DNSCrypt v2 doc.

Maybe asking them directly may help clarify this issue :-)
